src/magic_folder/uploader.py

```diff
@@ -410,9 +426,12 @@ def startService(self):
         Start UploaderService and initiate a periodic task
         to poll for LocalSnapshots in the database.
         """
-
         service.Service.startService(self)
 
+        # if we started with any local snapshots already in the
+        # database we must reflect this in our status service.
+        self._remote_snapshot_creator.initialize_upload_status()
```

Comment:
We should really wait to send status to any client, until we've updated the status with the appropriate information. I guess right now, we don't yield to the reactor between starting the web service and this call, but I think it would be better to explicitly ensure this is the case. (Particularly since I'm intending to make starting the magic folders asynchronous as part of #420, so we can check the on-grid version first).

My initial thought is that the `MagicFolder.ready` deferred should wait on each component service being ready (which would include populating status), and then `MagicFolderService` would notify the status service once all those have fired. That said, the code here *does* shorten the window where we could present stale data, so this can be handled in a followup (though I wouldn't consider #517 fixed until that is done).